### DISCUSSION TITLE
GTK dark theme/style not applying?

### DISCUSSION THREAD

**Discussion by peterrus** [regolith-linux/regolith-desktop#1044] [General]
**Edit: Solved in https://github.com/orgs/regolith-linux/discussions/1044#discussioncomment-10691886**

Hi! I am running Ubuntu 24.04 with Regolith 3.2 and I having trouble getting the dark theme to apply to Nautilus, gnome-settings and all other GTK applications. Even though the dark theme/style is applied all GTK windows keep using the light/default theme (On i3 and sway sessions) as seen in screenshot below. When starting a Gnome (Wayland or X11) session everything works as expected.

Am I doing something wrong or should I open an issue for this?

This happens on a clean fully updated install of both Ubuntu+Regolith.

*Edit*: I think GTK2/3 windows are styled properly and it's just GTK4 stuff that doesn't respect the setting under Regolith sessions.

*Edit2*: Third party (non-gtk)-applications that use the 'system style' to determine if they should show a dark/light theme also default to the light style even thought I set the system style to dark in gnome-settings.

![Screenshot from 2024-08-19 13-09-46](https://github.com/user-attachments/assets/c7a58ff6-ea93-4129-b6ac-d65f4fb8dfdd)

**Comment by peterrus** [comment]
@kgilmer Is this something that you have seen on your dev environments? Should I open an issue for this?

**Reply by kgilmer** [reply]
Hi @peterrus , yes this is a known issue.  Gtk4 apps in general do not theme well in regolith.  "Dark mode" is a gnome feature that is unimplemented in regolith as of version 3.2.

**Reply by peterrus** [reply]
Ah! That's not the news I was hoping for but good to know this is at least to be expected ;)

Is this documented somewhere (which I missed)? If not: At least we now have this thread.

**Reply by made4this** [reply]
@peterrus, @kgilmer: [James Thistlewood](https://github.com/i3/i3/discussions/5896#discussioncomment-8556941) solved it for us!

**Reply by peterrus** [reply]
That's great! Works for me! Thanks :+1:

**Comment by fjellvannet** [comment]
Upvote for this issue :)

Or for creating an issue of this if it has not been created yet.

**Reply by peterrus** [reply]
See https://github.com/orgs/regolith-linux/discussions/1044#discussioncomment-10691886

**Comment by khos2ow** [comment]
For visibility, solution suggested by @made4this solves the issue.

Create `$HOME/.config/xdg-desktop-portal/portals.conf` with contents:

```
[preferred]
default=gtk;wlr
```

And restart.

**Comment by maynouf** [comment]
I had the same problem with Linux Mint 22 with Cinnamon. Keep in mind that without a GNOME/XFCE session, GTK apps have nothing to read theme settings from. 

`xsettingsd` is a lightweight daemon that fills that role:

```sudo apt install xsettingsd```

Then you can create `~/.config/xsettingsd/xsettingsd.conf` with:
```
Net/ThemeName "Adwaita-dark"
Net/IconThemeName "Adwaita"
```
(keep in mind that you've got to have those themes installed)

Check if it worked with:

`xsettingsd &`

To make it permanent, make a file with `exec --no-startup-id xsettingsd` in `~/.config/regolith3/i3/config.d/xsettingsd.conf`.
